coil/src/main/java/dev/chrisbanes/accompanist/coil/Coil.kt

```diff
@@ -69,54 +71,41 @@ import coil.request.ImageResult
  * @param shouldRefetchOnSizeChange Lambda which will be invoked when the size changes, allowing
  * optional re-fetching of the image. Return true to re-fetch the image.
  * @param onRequestCompleted Listener which will be called when the loading request has finished.
- * @param error Content to be displayed when the request failed.
- * @param loading Content to be displayed when the request is in progress.
- * @param image Content to be displayed when the request is successful.
+ * @param content Content to be displayed for the given state.
  */
 @Composable
 fun CoilImage(
     data: Any,
     modifier: Modifier = Modifier,
     imageLoader: ImageLoader = Coil.imageLoader(ContextAmbient.current),
-    shouldRefetchOnSizeChange: (currentResult: RequestResult, size: IntSize) -> Boolean = defaultRefetchOnSizeChangeLambda,
-    onRequestCompleted: (RequestResult) -> Unit = emptySuccessLambda,
-    error: @Composable ((ErrorResult) -> Unit)? = null,
-    loading: @Composable (() -> Unit)? = null,
-    image: @Composable (SuccessResult) -> Unit
+    shouldRefetchOnSizeChange: (currentResult: CoilImageState, size: IntSize) -> Boolean = defaultRefetchOnSizeChangeLambda,
+    onRequestCompleted: (CoilImageState) -> Unit = emptySuccessLambda,
+    content: @Composable (state: CoilImageState) -> Unit
```

Comment:
what about naming the param `imageState` — state is a pretty overloaded term.

Reply:
Done